### ISSUE TITLE
Can support Optimize Imports?

### ISSUE THREAD

**Issue by Misaka15535** [intellij-rust/intellij-rust#8630]
## Environment

* **IntelliJ Rust plugin version:0.4.165.4438-213**
* **Rust toolchain version:1.61.0-nightly**
* **IDE name and version:CLion 2021.3.3 Build #CL-213.6777.58, built on February 1, 2022**
* **Operating system:win 10**

## Problem description
I don't know if this feature plugin was originally supported, but I hope I can use it.
This is useful for refactoring code.

## Steps to reproduce
Create a new file and copy useless imports from other files.
Then press the “Optimize Imports” shortcut key.
They don't get any effect.
![image](https://user-images.githubusercontent.com/46892084/156696957-9c498bcb-102a-4240-9b9f-e7604c6e86b9.png)

**Comment by Kobzol** [comment]
There is an inspection for detecting unused imports, however it's currently disabled because it has some false positives. You can enable it in `Settings -> Editor -> Inspections -> Rust -> Lints -> Unused import`. With the inspection enabled, `Optimize imports` should remove the unused imports.

**Comment by Misaka15535** [comment]
Thank you. It's ready to use now.

**Comment by Misaka15535** [comment]
But it still doesn't work for the following example.

```use crate::Gl;```

This may be a known defect now. Is support for that content currently planned?

**Comment by Kobzol** [comment]
The false positives are mostly related to imported things that are used within macros or attributes. Is `crate::Gl` resolved? If it's unresolved, it won't be marked as unused.

**Comment by Misaka15535** [comment]
sorry.I don't understand the meaning of this sentence."Is crate::Gl resolved"

**Comment by Kobzol** [comment]
I meant if the plugin was able to resolve the import to some existing item in your codebase. If you `Ctrl + Click` (or right click -> Go to definition) on the `Gl` part of the import, does it lead you to some item (trait/struct/...)?

**Comment by Misaka15535** [comment]
Your guess is right, plugins don't recognize this.
So this might be an unsolvable problem.

**Comment by Undin** [comment]
> Your guess is right, plugins don't recognize this.

@Misaka15535 Do I understand correctly that it's the same problem as described in #8620 ?

**Comment by Misaka15535** [comment]
> > Your guess is right, plugins don't recognize this.
> 
> @Misaka15535 Do I understand correctly that it's the same problem as described in #8620 ?

I'm not so sure, maybe after #8620 is fixed, this will work. But what if I remove the reference to the package too? This seems to still lead to unrecognized.

**Comment by Undin** [comment]
@Misaka15535 I've build the plugin with the changes from #8631
So, you can check if everything works as you expected

[intellij-rust-0.4.167.SNAPSHOT-213-dev.zip](https://github.com/intellij-rust/intellij-rust/files/8198412/intellij-rust-0.4.167.SNAPSHOT-213-dev.zip)

**Comment by Misaka15535** [comment]
@Undin  

build Sync fail.

```
C:/Users/Admin/.cargo/bin/cargo.exe check --message-format json --workspace --all-targets
error: failed to run `rustc` to learn about target-specific information
error: failed to run `rustc` to learn about target-specific information
Caused by:
  process didn't exit successfully: `C:\Users\Admin\AppData\Roaming\JetBrains\CLion2021.3\plugins\intellij-rust\bin\windows\x86-64\intellij-rust-native-helper.exe C:\Users\Admin\.cargo\bin\rustc.exe - --crate-name ___ --print=file-names --crate-type bin --crate-type rlib --crate-type dylib --crate-type cdylib --crate-type staticlib --crate-type proc-macro --print=sysroot --print=cfg` (exit code: 0xc0000135, STATUS_DLL_NOT_FOUND)
```
![image](https://user-images.githubusercontent.com/46892084/157155635-fcf8e886-d6d5-4a1c-b0e5-2e0f035326f2.png)


But I can run it successfully on clion terminals.
![image](https://user-images.githubusercontent.com/46892084/157155584-8ef55612-df94-4d93-81e2-14e433019eda.png)
